ledger/test-common/src/main/scala/com/digitalasset/platform/testing/PromiseElementObserver.scala

```diff
@@ -0,0 +1,28 @@
+// Copyright (c) 2022 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.platform.testing
+
+import io.grpc.stub.StreamObserver
+
+import scala.concurrent.Promise
+
+/** Implementation of [[StreamObserver]] designed to expose a promise of a single element
+  */
+private[testing] final class PromiseElementObserver[A] extends StreamObserver[A] {
+
+  val promise: Promise[A] = Promise[A]()
+
+  override def onNext(value: A): Unit = synchronized {
```

Comment:
This observer doesn't complete the underlying `StreamObserver` when the first element appears (see how `SizeBoundObserver` does it). I also think you can actually use that already existing observer instead of rolling this one. wdyt?

Reply:
There is no `underlying StreamObserver` in `PromiseEelementObserver` and I do not see it is required. Also `SizeBoundObserver` does not expose a promise, or you want to extend it to include that too?

Reply:
I just realised I don't need `synchronised` as I only operate with safe `Promise`

Reply:
> `SizeBoundObserver` does not expose a promise

Why do you need a promise ? Future has `isCompleted` too, in case that was the issue. You can just use the overloaded `flatTransactions` method with the `take=1` parameter and use the result like you used the promise.

Reply:
oh wow, you're right, my solution was an overkill :) thanks!